### ISSUE TITLE
Parsing bug with concise generator methods

### ISSUE THREAD

**Issue by humanchimp** [babel/babel#912]
I ran into a parsing bug while porting a project from traceur to babel, and managed to simplify it to this:

```javascript
({ *a(a) { a.b; } }) //  Cannot read property 'push' of undefined` 
```

It's not a problem for non-generator functions. I think shadowing like this is bad style, but is supported by the spec.

**Comment by sebmck** [comment]
Fixed as of 4.6.1, thanks!

**Comment by humanchimp** [comment]
:+1: Thank you!
